### DISCUSSION TITLE
Eigenvalues and vectors

### DISCUSSION THREAD

**Discussion by RichardBJ** [pymc-devs/pytensor#480] [Q&A]
Hi there, was trying to construct a pymc model that calculates eigenvectors in the middle (and other matrix transforms).  PYMC is now using this pytensor library for tensor operations I believe, so I dont think there is a way to do it with pytensor but thought I should check?
Kindest Regards
Richard

**Comment by ricardoV94** [comment]
We have something called `eigvalsh` not sure if that's all you need: https://github.com/pymc-devs/pytensor/blob/7bb18f3a3590d47132245b7868b3a4a6587a4667/pytensor/tensor/slinalg.py#L553-L554

You may have more luck getting a reply if you post on our Discourse forum: https://discourse.pymc.io/

**Reply by RichardBJ** [reply]
Gosh I'm sorry for the delay in reply. And thanks for the suggestion, in fact, I think my problem has an easier solution which I now have running which leaving my matrix code as is and then treating it as a "a “black box” likelihood function" as exemplared...
[here](https://www.pymc.io/projects/examples/en/latest/case_studies/blackbox_external_likelihood_numpy.html)
Thanks!
R.
